src/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.tsx

```diff
@@ -0,0 +1,194 @@
+import Box from '@mui/material/Box'
+import { FC, SyntheticEvent } from 'react'
+import { useTranslation } from 'react-i18next'
+import AppButton from '~/components/app-button/AppButton'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import { styles } from '~/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.styles'
+import {
+  ButtonVariantEnum,
+  ComponentEnum,
+  ProfessionalCategory,
+  ProfessionalSubject,
+  ProficiencyLevelEnum
+} from '~/types'
+import useForm from '~/hooks/use-form'
+import AppAutoComplete from '~/components/app-auto-complete/AppAutoComplete'
+import AddIcon from '@mui/icons-material/Add'
+import { professionalSubjectTemplate } from '~/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.constants'
+import ProficiencyLevelSelect from '~/containers/proficiency-level-select/ProficiencyLevelSelect'
+
+interface ValueWithHandler<T> {
+  value: T
+  handleChange: (value: T) => void
+}
+
+interface SubjectGroupProps {
+  isCategoryDisabled: boolean
+  subjectName: ValueWithHandler<string | null>
+  proficiencyLevels: ValueWithHandler<ProficiencyLevelEnum[]>
+}
+
+function SubjectGroup({
+  isCategoryDisabled,
+  subjectName,
+  proficiencyLevels
+}: SubjectGroupProps) {
+  const { t } = useTranslation()
+
+  return (
+    <Box sx={styles.item}>
+      <Box sx={styles.checkboxGroup}>
+        <Checkbox checked disabled={isCategoryDisabled} sx={styles.checkbox} />
+        <AppAutoComplete
+          fullWidth
+          onChange={(_, value) => subjectName.handleChange(value)}
+          options={['Subject1', 'Subject2', 'Subject3']} // TODO: replace with actual subjects from backend
+          textFieldProps={{
+            label: `${t('editProfilePage.profile.professionalTab.subject')}*`
+          }}
+          value={subjectName.value}
+        />
+      </Box>
+      <ProficiencyLevelSelect
+        fullWidth
+        label={t('editProfilePage.profile.professionalTab.proficiencyLevels')}
+        onChange={(event) =>
+          proficiencyLevels.handleChange(
+            event.target.value as ProficiencyLevelEnum[]
+          )
+        }
+        value={proficiencyLevels.value}
+      />
+    </Box>
+  )
+}
+
+interface AddProfessionalCategoryModalProps {
+  closeModal: () => void
+  initialValues?: ProfessionalCategory
+}
+
+const AddProfessionalCategoryModal: FC<AddProfessionalCategoryModalProps> = ({
+  closeModal,
+  initialValues: initialValuesFromProps
+}) => {
+  const { t } = useTranslation()
+
+  const initialFormValues = {
+    mainStudyCategory: initialValuesFromProps?.name ?? '',
+    subjects: initialValuesFromProps?.subjects ?? [professionalSubjectTemplate]
+  }
+
+  // TODO: add more validations if needed
+  const { data, errors, handleDataChange, handleSubmit } = useForm({
+    initialValues: initialFormValues,
+    // eslint-disable-next-line
```

Comment:
It is mock code because `useForm` requires `onSubmit` to be an `async` function and eslint requires to have at least one `await` inside. As it is mock data, we don't have `await` yet and I decided to use `eslint-disable-next-line` rule